### PR TITLE
feat: migrate kafka consumer bridge to source + connector

### PR DESCRIPTION
**Pull request by thalesmg**
Fixes https://emqx.atlassian.net/browse/EMQX-11848

Release version: e5.6

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_bridge/src/emqx_bridge_v2_api.erl [hunk 1]
```diff
@@ -775,7 +775,7 @@ handle_update(ConfRootKey, Id, Conf0) ->
         Id,
         case emqx_bridge_v2:lookup(ConfRootKey, BridgeType, BridgeName) of
             {ok, _} ->
-                RawConf = emqx:get_raw_config([bridges, BridgeType, BridgeName], #{}),
```
Comment (thalesmg) [inline apps/emqx_bridge/src/emqx_bridge_v2_api.erl:778]: :see_no_evil:

File: apps/emqx_connector/src/schema/emqx_connector_schema.erl [hunk 2]
```diff
@@ -205,7 +207,11 @@ bridge_configs_to_transform(
                 emqx_utils_maps:deep_get(
                     [<<"actions">>, to_bin(BridgeType), to_bin(BridgeName)],
                     RawConfig,
-                    undefined
```
Comment (thalesmg) [inline apps/emqx_connector/src/schema/emqx_connector_schema.erl:208]: Fixing this: https://github.com/emqx/emqx/pull/12486#issuecomment-1932720408

File: apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_consumer.erl [hunk 3]
```diff
@@ -183,29 +195,91 @@ on_stop(ResourceId, _State = undefined) ->
...
+    ),
     stop_client(ClientID),
+    ?tp(kafka_consumer_subcriber_and_client_stopped, #{instance_id => ConnectorResId}),
     ok.
 
--spec on_get_status(resource_id(), state()) -> connected | disconnected.
-on_get_status(_ResourceID, State) ->
+-spec on_get_status(resource_id(), connector_state()) -> connected | disconnected.
+on_get_status(_ResourceID, _State) ->
+    ?status_connected.
```
Comment (thalesmg) [inline apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_consumer.erl:215]: I didn't see an API for health checking just the connection without knowing a topic or subscription group...  :see_no_evil:

Comment (zhongwencool) [reply apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_consumer.erl:215]: Maybe we can use `brod_sup:find_client(Client)` to check the client is exist .

File: apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_consumer.erl [hunk 4]
```diff
@@ -583,3 +682,19 @@ infer_client_error(Error) ->
...
+        ConnectorResId, ?kafka_subscriber_ids, AllocatedSubscriberIds
+    ).
+
+deallocate_subscriber_id(ConnectorResId, SubscriberId) ->
+    AllocatedResources = emqx_resource:get_allocated_resources(ConnectorResId),
+    AllocatedSubscriberIds0 = maps:get(?kafka_subscriber_ids, AllocatedResources, []),
+    AllocatedSubscriberIds = AllocatedSubscriberIds0 -- [SubscriberId],
+    ok = emqx_resource:allocate_resource(
+        ConnectorResId, ?kafka_subscriber_ids, AllocatedSubscriberIds
+    ).
```
[the remark this answers is not in the document]
Comment (thalesmg) [reply apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_consumer.erl:700]: It's a better idea, indeed.

File: apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_consumer.erl (hunk 3 above)
Comment (thalesmg) [reply apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_consumer.erl]: :heavy_check_mark:

File: apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_consumer.erl (hunk 4 above)
[the remark this answers is not in the document]
Comment (thalesmg) [reply apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_consumer.erl]: :heavy_check_mark:

**Review by zhongwencool — APPROVED** [review]

**Pull request merged**